controller/src/main/java/io/pravega/controller/server/eventProcessor/LocalController.java

```diff
@@ -565,6 +566,23 @@ private StreamSegments getStreamSegments(List<SegmentRange> ranges) {
                 .thenApply(status -> ModelHelper.encode(status.getState(), stream + " " + txnId));
     }
 
+    @Override
+    public AsyncIterator<TransactionInfo> listCompletedTransactions(Stream stream) {
+        // return controller.listCompletedTxns(stream.getScope(), stream.getStreamName(), requestIdGenerator.nextLong());
+
+        final Function<String, CompletableFuture<Map.Entry<String, Collection<TransactionInfo>>>> function = token ->
+                controller.listCompletedTxns(stream.getScope(), stream.getStreamName(), PAGE_LIMIT, token, requestIdGenerator.nextLong())
+                        .thenApply(result -> {
+                            List<TransactionInfo> txnInfoList = new ArrayList<>();
+                            result.getKey().forEach((id, status) -> {
+                                txnInfoList.add(new TransactionInfo(id, Transaction.Status.valueOf(status.name())));
+                            });
+                            return new AbstractMap.SimpleEntry<>(result.getValue(), txnInfoList);
```

Comment:
This should be covered by a test.

Reply:
Added a testcase to cover this code changes.